### ISSUE TITLE
Feature request: Support ui-related query params in addon-links

### ISSUE THREAD

**Issue by jfsiii** [storybooks/storybook#3787]
related to https://github.com/storybooks/storybook/issues/1739

The storybook URL has query params to put the UI in a particular state, (e.g. nav/addon panel appearance, position, focus, etc), but `addon-links` only supports the `selectedStory` and `selectedKind` params (via `story` & `kind`).

e.g. [story with no panels shown](https://storybooks-official.netlify.com/?selectedKind=Components%7CForm%2FButton&selectedStory=with%20onclick&full=0&addons=0&stories=0&panelRight=0&addonPanel=storybook-addon-viewport%2Faddon-panel) and [same story w/background panel shown & focused](https://storybooks-official.netlify.com/?selectedKind=Components%7CForm%2FButton&selectedStory=with%20onclick&full=0&addons=1&stories=0&panelRight=0&addonPanel=storybook-addon-background%2Fbackground-panel)

I'd like everything that is available in the URL to be available to [`hrefTo`](https://github.com/storybooks/storybook/tree/master/addons/links#hrefto-function) and/or [`<LinkTo/>`](https://github.com/storybooks/storybook/tree/master/addons/links#linkto-component-react-only)

This way one story could link to another story in a particular state. e.g.

e.g.

```jsx
<LinkTo
  kind="Components|Form/Button"
  story="with onclick"
  addons
  stories=0 // or `false`, etc
  panelRight=0 // or `false`, etc
  addonPanel="storybook-addon-background/background-panel"
>Story with background panel focused</LinkTo>
```

would produce a relative URL like

```
?selectedKind=Components%7CForm%2FButton&selectedStory=with%20onclick&full=0&addons=1&stories=0&panelRight=0&addonPanel=storybook-addon-background%2Fbackground-panel
```

**Comment by arshadkazmi42** [comment]
I would like to take this up.

**Comment by igor-dv** [comment]
@arshadkazmi42 feel free to PR.

**Comment by ndelangen** [comment]
I'll see what I can do, I need to fix addon-links in #4086 anyway

**Comment by airtonix** [comment]
what happened with this ?
